src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -1307,10 +1385,12 @@ def universal_undo_preprocessing_fn(inputs: torch.Tensor) -> np.ndarray:
         inputs = np.ascontiguousarray(inputs, dtype=np.uint8)
         return inputs
 
-    def process_extreme_batch(self) -> Tuple[np.ndarray, np.ndarray]:
+    def process_extreme_batch(self) -> np.ndarray:
         """
-        Processes the extreme batch, and returns 2 image batches for visualization - one with predictions and one with GT boxes.
-        :return:Tuple[np.ndarray, np.ndarray], the predictions batch, the GT batch
+        Processes the extreme batch, and returns list of images for visualization.
+        Default implementations stacks GT and prediction overlays horisontally.
+
+        :return: np.ndarray A 4D tensor of [BHWC] shape with visualizations of the extreme batch.
```

Comment:
Mainly because we are using channels last layout to draw lines/boxes using cv2. So we have to move BCHW -> BHWC in any case. But then there is no reason to move it back to BCHW since both TB & W&B works perfectly fine with `data_format="NHWC"` 
So we can call it optimization / removal of reduntant data copy.